app/src/main/java/org/p2p/wallet/auth/web3authsdk/GoogleSignInHelper.kt

```diff
@@ -52,10 +64,18 @@ class GoogleSignInHelper() {
             getSignInClient(context).getSignInCredentialFromIntent(result.data)
         } catch (ex: ApiException) {
             if (ex.statusCode != USER_CANCELED_DIALOG_CODE) {
-                Toast.makeText(context, ex.toString(), Toast.LENGTH_SHORT).show()
+                if (connectionStateProvider.hasConnection()) {
+                    view?.showErrorSnackBar(ex.message ?: ex.toString())
```

Comment:
i would rather pass callback or idk, instead of working with view
Helper should be stupid i think

Reply:
i can make it in a callback
i do not mind

Reply:
well, i meant pass callback as a parameter really :) 
It would leave us with only one method needed instead of three: attach / method that we need / detach.
this all detach/attach thing is not needed here - listener / handler is not Context-aware thing, there will be no leak.

Reply:
Fuuuuuu! memory leak